prusti-specs/src/export_spec_rewriter.rs

```diff
@@ -0,0 +1,334 @@
+    let mut type_path = syn::Path {
+        leading_colon: None,
+        segments: syn::punctuated::Punctuated::new(),
+    };
+    get_prusti_path_in_attr!(item_impl, "type_path", type_path);
+
+    let macro_content =
+        expand_type_path_in_macro(&mod_path, quote!(#item_impl), uses_same_level, &type_path);
+    let item_ty = &item_impl.self_ty;
+    if let syn::Type::Path(type_path) = item_ty.as_ref() {
+        path.segments.extend(type_path.path.segments.clone());
+    }
+
+    for item in item_impl.items.iter() {
+        match item {
+            syn::ImplItem::Method(item_method) => {
+                rewrite_method(&path, item_method, &mod_path, macros_tokens);
+            }
+            _ => {
+                return Err(syn::Error::new(
+                    item.span(),
+                    "expected a method".to_string(),
+                ));
+            }
+        }
+    }
+
+    get_prusti_path_in_attr!(item_impl, "export_path", path);
+    parse_valid_export_type(&mut path);
+    let macro_ident = generate_macro_ident(path);
+    macros_tokens.extend(generate_macro_for_export(macro_ident, macro_content));
+    Ok(())
+}
+
+fn rewrite_method(
+    path: &syn::Path,
+    item_method: &syn::ImplItemMethod,
+    impl_mod_path: &syn::Path,
+    macros_tokens: &mut TokenStream,
+) {
+    let mut path = path.to_owned();
+
+    let mut type_path = syn::Path {
+        leading_colon: None,
+        segments: syn::punctuated::Punctuated::new(),
+    };
+
+    get_prusti_path_in_attr!(item_method, "type_path", type_path);
+    let macro_content = expand_type_path_in_macro(
+        impl_mod_path,
+        quote!(
+            impl #type_path {
+                #item_method
+            }
+        ),
+        &TokenStream::new(),
+        &type_path,
+    );
+    path.segments.push(syn::PathSegment {
+        ident: item_method.sig.ident.clone(),
+        arguments: syn::PathArguments::None,
+    });
+
+    get_prusti_path_in_attr!(item_method, "export_path", path);
+    parse_valid_export_type(&mut path);
+    let macro_ident = generate_macro_ident(path);
+    macros_tokens.extend(generate_macro_for_export(macro_ident, macro_content));
+}
+
+pub fn parse_valid_export(attr: TokenStream) -> syn::Result<syn::Path> {
+    let span = attr.span();
+    match attr.is_empty() {
+        true => Err(syn::Error::new(span, "export path must be non-empty")),
+        false => match attr.to_string().trim() == "::" {
+            true => Ok(syn::Path {
+                leading_colon: None,
+                segments: syn::punctuated::Punctuated::new(),
+            }),
+            false => syn::parse2(attr).map_err(|_e| syn::Error::new(span, "invalid export path")),
+        },
+    }
+}
+
+fn parse_valid_export_type(ty_path: &mut syn::Path) {
+    for seg in &mut ty_path.segments {
+        if let syn::PathArguments::AngleBracketed(_) = &mut seg.arguments {
+            seg.arguments = syn::PathArguments::None;
+        }
+    }
+}
```

Comment:
```suggestion
}

```

Reply:
Please go through the file diff here on github and make sure every file you added or modified is still ending with a linebreak. GitHub highlights when it doesn't with a red symbol.